ref: Add ruff rules for bugbear (B)

cbornet opened the pull request.

cbornet commented on src/backend/base/langflow/custom/custom_component/base_component.py:
```diff
@@ -35,12 +35,12 @@ def __init__(self, **data):
 
     def __setattr__(self, key, value):
-        if key == "_user_id" and hasattr(self, "_user_id") and getattr(self, "_user_id") is not None:
+        if key == "_user_id" and self._user_id is not None:
```
`_user_id` is a class attribute so it seems it will always be there ?

ogabrielluiz replied:
Agree.

cbornet commented on src/backend/base/langflow/graph/graph/state_model.py:
```diff
@@ -63,6 +63,6 @@ def create_state_model_from_graph(graph: BaseModel) -> type[BaseModel]:
     fields = {
         camel_to_snake(vertex.id): state_model_getter
-        for vertex, state_model_getter in zip(graph.vertices, state_model_getters)
+        for vertex, state_model_getter in zip(graph.vertices, state_model_getters, strict=False)
```
Reading the code, it seems `graph.vertices` and `state_model_getters` could have distinct size. Is it intended ?

ogabrielluiz replied:
state_model_getters is built by iterating graph.vertices

ogabrielluiz replied:
Yeah. We probably need to add the same checks on the second iteration or just build the fields dict in one go.

ogabrielluiz replied:
This part is still highly experimental and requires a lot more testing/inclusion into features

cbornet commented on src/backend/base/pyproject.toml:
```diff
@@ -151,6 +151,7 @@ line-length = 120
 [tool.ruff.lint]
 select = [
     "ASYNC",
+    "B904",
```
Not enforcing all B rules atm. Only B904, the one for `raise SomeError from exc`

cbornet commented:
@ogabrielluiz thanks for the review.
I modified the PR to fix all B rules.
In retrospect, I should have done it in another commit instead of amending the first one... Sorry for that.
Can you take a look again ?
The main changes are that I replaced `warnings.warn` by `logger.warning` which is more appropriate (see https://docs.python.org/3/howto/logging.html#when-to-use-logging)

ogabrielluiz reviewed (approved):
LGTM
